### ISSUE TITLE
Create a new MSI (WiX V4) project results in Error: Exeption of type 'System.Exception' was trown.

### ISSUE THREAD

**Issue by rspcoding** [firegiant/HeatWaveSupport#48]
### HeatWave Version

1.0.2

### Visual Studio Version

17.6.8

### Repro Steps

1. HeatWave 1.0.2 installed on Visual Studio 2022 Enterprise 17.6.8
2. Restart of the System
3. Try to create a new MSI Package (WiX v4) in Visual Studio

System:
Windows 10 22H2 (19045.3570) VDI (Offline System with no Internet access)

### Actual Result

After klick button [Create] I get an Error: Exeption of type 'System.Exception' was trown.

![image](https://github.com/firegiant/HeatWaveSupport/assets/11905685/cebd9701-f2b1-4123-bb8d-7144a0f1541e)

**Comment by firegiantco** [comment]
We are not able to reproduce this issue.  Does it only happen in an offline system for you or can you reproduce it when on a connected system?

**Comment by jcapriot** [comment]
I am also having this same issue. I am unable to create any project using the Heatwave Extension,
### HeatWave Version
1.0.2.1
### Visual Studio Version
17.8.5

I have been trying (but failing) to find any more relevant debug information to provide you with more details about the bug report, but literally the only thing is the message reported above.

Some things I've tried

- Uninstall/Reinstall
- Start VS with administrative privileges
- Updated Visual Studio
- Placing the solution and project in different directories
    - This successfully creates a folder with the Solution Name
    - Creates a single hidden file in a hidden folder for the solution `.vs\Solution Name\v17\.suo`
    - Creates a solution file with:
```Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 17
VisualStudioVersion = 17.8.34511.84
MinimumVisualStudioVersion = 10.0.40219.1
Global
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {40E0039B-998E-4D2F-B63B-8F8DD76A364E}
	EndGlobalSection
EndGlobal
```
    - Does not create any other files or directories

- Placing the solution and project in the same directories
    - This doesn't create any files or folders.
    
### System:
Windows 10 V22H2


This is a list of all of my installed products in Visual Studio:
```
Microsoft Visual Studio Community 2022
Version 17.8.5
VisualStudio.17.Release/17.8.5+34511.84
Microsoft .NET Framework
Version 4.8.09037

Installed Version: Community

Visual C++ 2022   00482-90000-00000-AA508
Microsoft Visual C++ 2022

ASP.NET and Web Tools   17.8.358.6298
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.8.358.6298
Azure App Service Tools v3.0.0

C# Tools   4.8.0-7.23572.1+7b75981cf3bd520b86ec4ed00ec156c8bc48e4eb
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

FireGiant HeatWave   1.0.2.1
Visual Studio tooling for WiX projects.  For more information about HeatWave, visit http://www.firegiant.com.

Intel® C++ Compiler   2024.0
Intel® C++ Compiler 2024.0, Copyright © 2002-2023 Intel Corporation. All rights reserved.

Intel® DPC++ Compatibility Tool   2024.0
Intel® DPC++ Compatibility Tool 2024.0, Copyright © 2019-2023 Intel Corporation. All rights reserved.

Intel® Fortran Compiler   2024.0
Intel® Fortran Compiler 2024.0, Copyright © 2002-2023 Intel Corporation. All rights reserved.

Intel® Inspector 2024   1.3.21
Intel® Inspector 2024, (Plugin Version 1.3.21),  Copyright © 2009-2023 Intel Corporation. All rights reserved.

Intel® Libraries for oneAPI   2024.0
Intel® Libraries for oneAPI 2024.0, Copyright © 2019-2023 Intel Corporation. All rights reserved.

Intel® oneAPI DPC++ Compiler   2024.0
Intel® oneAPI DPC++ Compiler 2024.0, Copyright © 2019-2023 Intel Corporation. All rights reserved.

Intel® oneAPI Menu & Samples   10.10.392.9731
Intel® oneAPI Visual Studio Menu & Samples Extension

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   17.0.20920.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.8.0-7.23572.1+7b75981cf3bd520b86ec4ed00ec156c8bc48e4eb
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.
```

**Comment by rspcoding** [comment]
> We are not able to reproduce this issue. Does it only happen in an offline system for you or can you reproduce it when on a connected system?

We tried it on another offline system (VM Ware Workstation) and it was possible to create a new project there.

The error is therefore due to the specific system, but we could not find any relevant logs.
Is there a special directory with relevant logs to find the error?

**Comment by firegiantco** [comment]
It feels like there is a missing component on the failing system.  You could run the Visual Studio Installer and from its main screen select `More -> Export Configuration`.  That will spit out a `.vsconfig` that you could compare between the two systems.

**Comment by talaviram** [comment]
Is there anywhere where it is written what are the required components ?

**Comment by cjt-westwoodps** [comment]
I'm having the same issue. 

**HeatWave Version**
1.0.2.1

**Visual Studio Version**
17.9.4

**System**
Windows 11 23H2

I've tried all the same things reported by others to no avail:
- Uninstall/Reinstall
- Start VS with administrative privileges
- Updated Visual Studio
- Placing the solution and project in different directories 

I can't find any relevant VS logs to indicate what the problem might be. I am able to create other solutions but nothing from the FireGiant WixToolset.

**Comment by FireGiantHelp** [comment]
The log file shows the expected HeatWave dependencies. There are many ways to configure workloads in Visual Studio so we are hoping that evaluating failing VS configurations will enable us to find possible missing dependencies.  Open the Visual Studio Installer, click `More` then `Export Configuration`) and attach the `.vsconfig` to this issue.

**Comment by fishnet37222** [comment]
I'm also seeing this issue.  I've attached my `.vsconfig` file as requested.
[vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14810230/vsconfig.txt)

**Comment by cjt-westwoodps** [comment]
Here is my .vsconfig file. Please let me know if I can provide any more information.
[vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14823963/vsconfig.txt)

**Comment by rspcoding** [comment]
Our Visual Studio has now been updated to 17.8.8 and I have installed the latest HeatWave version 1.0.3.
However, the error still persists.

Attached is the .vsconfig file from my system.
[vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14858168/vsconfig.txt)

**Comment by Strahbehry** [comment]
Same issue with 17.9.5

[vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14907097/vsconfig.txt)

**Comment by TheCakeIsNotALie** [comment]
Hi, I am encountering the same issue after updating from 17.6.x to 17.9.5 with Heatwave 1.0.2.1

[vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14916157/vsconfig.txt)

I also dug in the ActivityLog.xml of Visual Studio and found this error detailing it (sorry as it is in french):
```
<entry>
    <record>909</record>
    <time>2024/04/09 08:45:39.742</time>
    <type>Error</type>
    <source>Template provider aggregator</source>
    <description>System.Exception: Une exception de type 'System.Exception' a été levée.
   à Microsoft.VisualStudio.CommonIDE.Solutions.Interop.IVsNativeEnvironmentInternal.WizLoadAndRunFile(String pwszWizard, String pwszDestination, Boolean bCloseSolution, Boolean bRunSilent, Array psaCustomParams, String pwszFrameworkVersion)
   à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   à Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
   à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.AddFromTemplateEx(String FileName, String Destination, String ProjectName, String SolutionName, Boolean Exclusive, UInt32 Options)
   à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<>c__DisplayClass25_0.<EnvDTE100.Solution4.AddFromTemplateEx>b__0()
   à Microsoft.VisualStudio.NewProjectDialog.LegacyVsTemplateProvider.GetInstantiationResult(CreateProjectResult result)
   à Microsoft.VisualStudio.NewProjectDialog.LegacyProjectTemplateProvider.<InstantiateTemplateAsync>d__10.MoveNext()
--- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
   à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   à Microsoft.VisualStudio.TemplateProviders.TemplateProviderSupport.ExceptionLoggingTemplateProviderDecorator.<InstantiateTemplateAsync>d__9.MoveNext()</description>
  </entry>
```

**Comment by TheCakeIsNotALie** [comment]
> Hi, I am encountering the same issue after updating from 17.6.x to 17.9.5 with Heatwave 1.0.2.1
> 
> [vsconfig.txt](https://github.com/firegiant/HeatWaveSupport/files/14916157/vsconfig.txt)
> 
> I also dug in the ActivityLog.xml of Visual Studio and found this error detailing it (sorry as it is in french):
> 
> ```
> <entry>
>     <record>909</record>
>     <time>2024/04/09 08:45:39.742</time>
>     <type>Error</type>
>     <source>Template provider aggregator</source>
>     <description>System.Exception: Une exception de type 'System.Exception' a été levée.
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Interop.IVsNativeEnvironmentInternal.WizLoadAndRunFile(String pwszWizard, String pwszDestination, Boolean bCloseSolution, Boolean bRunSilent, Array psaCustomParams, String pwszFrameworkVersion)
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
> --- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
>    à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
> --- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
>    à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<AddFromTemplateExAsync>d__248.MoveNext()
> --- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
>    à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    à Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
>    à Microsoft.VisualStudio.Threading.JoinableTask`1.CompleteOnCurrentThread()
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.AddFromTemplateEx(String FileName, String Destination, String ProjectName, String SolutionName, Boolean Exclusive, UInt32 Options)
>    à Microsoft.VisualStudio.CommonIDE.Solutions.Dte.DteSolution.<>c__DisplayClass25_0.<EnvDTE100.Solution4.AddFromTemplateEx>b__0()
>    à Microsoft.VisualStudio.NewProjectDialog.LegacyVsTemplateProvider.GetInstantiationResult(CreateProjectResult result)
>    à Microsoft.VisualStudio.NewProjectDialog.LegacyProjectTemplateProvider.<InstantiateTemplateAsync>d__10.MoveNext()
> --- Fin de la trace de la pile à partir de l'emplacement précédent au niveau duquel l'exception a été levée ---
>    à System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    à System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    à Microsoft.VisualStudio.TemplateProviders.TemplateProviderSupport.ExceptionLoggingTemplateProviderDecorator.<InstantiateTemplateAsync>d__9.MoveNext()</description>
>   </entry>
> ```

Follow up, upgrading to Heatwave 1.0.3 fixed the issue.

**Comment by Strahbehry** [comment]
Heatwave 1.0.3 doesn't work for me with 17.9.5 weirdly enough though

**Comment by FireGiantHelp** [comment]
We just released HeatWave v1.0.4 and this issue should be fixed there. If an exception is thrown, it is most likely a new issue so please open a new issue and attach Visual Studio's ActivityLog.xml to provide enough context to differentiate issues.
